app/models/invoice.rb

```diff
@@ -3,4 +3,11 @@ class Invoice < ApplicationRecord
   enum status: ["in progress", "completed", "cancelled"]
   has_many :invoice_items
   has_many :items, through: :invoice_items
+
+  def self.merchants_invoices(merchant)
+
+    joins(:invoice_items, :items).where(items: {merchant_id: merchant.id})
```

Comment:
Awesome activerecord query!! could this also go in merchant as an instance method ? items.joins(:invoice_items)?